### PR TITLE
add support for removal of default execution domain on AWS

### PR DESCRIPTION
**Pull request by MichielBijland**

### CODE REVIEW

**Review by davidweterings — COMMENTED** [review]
Looks good to me, but maybe we can add an explicit assert https://github.com/labd/mach-composer/blob/master/tests/unittests/terraform/test_endpoints.py#L7 here for the enable cdn logic?

**Comment by tleguijt** [comment]
I think instead of a setting we could make this a sane default?
And the same could be implemented for Azure so instead of only implementing this for AWS, shall we implement it for Azure as well.

So I think we have a couple of options;
- Remove the setting and just implement it as default behaviour
- Introduce a setting (so not AWS-specific) called `always_include_default_domain` which is set to `false` by default

**Comment by MichielBijland** [comment]
@tleguijt I'm all for sane defaults, but the current default is false which is backwards compatible but gives the option to remove it for compliance reasons. I rather have a good default and the ability to override it.

if you look at https://github.com/labd/mach-composer/blob/master/src/mach/templates/partials/endpoints/aws_endpoints.tf#L109 it's the same thing, we set a default but with no ability to change it but it will fail **most** compliance checks

On the azure bit, I don't think something like this exists on frontdoor.

**Comment by tleguijt** [comment]
We manage the frontdoor domains ourselves, so with this setting in mind we can choose to create / not to create the default Azure endpoint.
So in the end; you'll have the same behaviour, but other way of implementing it.
Therefore I think it would be a good idea to make this a generic setting.

And the example you brought up; if it fails compliance checks let's change the default there.
Also for this setting; if we make the default behaviour so that if you have a custom domain, the default domain won't be created anymore, we can include that in the Upgrade notes. If you want backwards compatibility, you can use the setting to re-enable it again.

**Comment by tleguijt** [comment]
So after discussing this with @MichielBijland came up with the following;

- By default, `disable_execute_api_endpoint` is set to `true` once an endpoint has a custom domain
- With this change the behaviour will be the same as for Azure; in the Azure frontdoor implementation we only route traffic from the default frontdoor domain to the endpoints if that endpoint has no custom domain.
- This is a breaking change and will be included in the Upgrade notes of the next version.
- An AWS-specific settings `always_include_default_domain` (or other name) is going to be added with which you can disable this behaviour for backwards compatability

File: docs/src/reference/syntax/sites.md [hunk 1]
```diff
@@ -54,11 +54,13 @@ aws:
   throttling_burst_limit: 5000
   throttling_rate_limit: 10000
   enable_cdn: true
+
 ```
 
 - `throttling_burst_limit` - Set burst limit for API Gateway endpoints
 - `throttling_rate_limit` - Set burst limit for API Gateway endpoints
 - `enable_cdn` - Defaults to false. Sets a CDN in front of this endpoint for better global availability. For AWS creates a CloudFront distribution
+- `disable_execute_api_endpoint` - Defaults to false and can't be set when using `enable_cdn`
```
Comment (tleguijt) [inline docs/src/reference/syntax/sites.md:63]: Could we make it a sane default and say, this is always disabled if you're using a custom domain?

**Comment by mvantellingen** [comment]
With all the recent changes this PR is now out of date.

Can you submit a new PR to the plugin at https://github.com/mach-composer/mach-composer-plugin-aws

**Pull request closed without merging**